x-pack/plugins/observability_solution/observability_ai_assistant/server/service/util/create_server_side_function_response_error.ts

```diff
@@ -19,15 +20,18 @@ export function createServerSideFunctionResponseError({
 }) {
   const isElasticsearchError = error instanceof errors.ElasticsearchClientError;
 
-  const sanitizedError: Record<string, unknown> = JSON.parse(
-    'toJSON' in error && typeof error.toJSON === 'function' ? error.toJSON() : JSON.stringify(error)
-  );
+  const sanitizedError: Record<string, unknown> = JSON.parse(JSON.stringify(error));
 
   if (isElasticsearchError) {
     // remove meta key which is huge and noisy
     delete sanitizedError.meta;
+  } else if (isAxiosError(error)) {
+    sanitizedError.response = { message: error.response?.data?.message };
+    delete sanitizedError.config;
   }
 
+  delete sanitizedError.stack;
```

Comment:
nit: I have an aversion against `delete` and prefer explicitly picking the properties we expect

Reply:
it would be `omit` in that case (no way for us to account for all the possible properties of this object). In this case I prefer `delete`ing properties of an object that is created a few lines up over having to re-assign stuff because of `omit`/`pick`.